actions/drain_node_handler.go

```diff
@@ -209,19 +214,46 @@ func (h *drainNodeHandler) deleteNodePods(ctx context.Context, log logrus.FieldL
 }
 
 func (h *drainNodeHandler) sendPodsRequests(ctx context.Context, pods []v1.Pod, f func(context.Context, v1.Pod) error) error {
-	batchSize := lo.Clamp(0.2*float64(len(pods)), 5, 20)
-	for _, batch := range lo.Chunk(pods, int(batchSize)) {
-		g, ctx := errgroup.WithContext(ctx)
-		for _, pod := range batch {
-			pod := pod
-			g.Go(func() error { return f(ctx, pod) })
-		}
-		if err := g.Wait(); err != nil {
-			return err
+	if len(pods) == 0 {
+		return nil
+	}
+
+	var (
+		parallelTasks = int(lo.Clamp(0.2*float64(len(pods)), 5, 20))
+		taskChan      = make(chan *v1.Pod, len(pods))
+		taskErrs      = make([]error, 0)
+		taskErrsMx    sync.Mutex
+		wg            sync.WaitGroup
+	)
+
+	h.log.Debugf("Starting %d parallel tasks for %d pods: [%v]", parallelTasks, len(pods), lo.Map(pods, func(t v1.Pod, i int) string {
+		return fmt.Sprintf("%s/%s", t.Namespace, t.Name)
+	}))
+
+	worker := func(taskChan <-chan *v1.Pod) {
+		for task := range taskChan {
+			if err := f(ctx, *task); err != nil {
+				taskErrsMx.Lock()
```

Comment:
maybe collect errors via error channel. There fore we can get rid-off both mutext and waitgroup. Just read all successes/errors.

Reply:
We'd still need waitgroup so we wait for all workers to finish (or we need to have a channel to signal when each one ends; I think waitgroup is simpler). Or maybe I am missing what you mean? 

For the errors, it can work either way; I thought mutex is simpler her since we only write to the slice in 1 place.